clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosRule.java

```diff
@@ -182,6 +189,9 @@ String apply(String[] params) throws IOException {
         if (result != null && NON_SIMPLE_PATTERN.matcher(result).find()) {
             throw new NoMatchingRule("Non-simple name " + result + " after auth_to_local rule " + this);
         }
+        if (toLowerCase && result != null) {
+            result = result.toLowerCase(Locale.ENGLISH);
```

Comment:
Do you know the range of characters permitted in kerberos names?

Reply:
I am not sure about the character range. I felt it is safe to set EN locale.  Hadoop implementation is also using EN locale.

@rajinisivaram  any thoughts here?